queries/java/locals.scm

```diff
@@ -1,26 +1,34 @@
 ; SCOPES
+; declarations
 (class_declaration
-   body: (_) @scope)
+  body: (_) @scope)
 (enum_declaration
-   body: (_) @scope)
-(method_declaration) @scope
+  body: (_) @scope)
+(method_declaration) @scope ; whole method_declaration because args
+
+; block
 (block) @scope
-(if_statement) @scope
+
+; if/else
+(if_statement) @scope ; if+else
 (if_statement
-  consequence: (_) @scope)
+  consequence: (_) @scope) ; if body in case there are no braces
 (if_statement
-  alternative: (_) @scope)
-(try_statement) @scope
-(catch_clause) @scope
-(do_statement) @scope
+  alternative: (_) @scope) ; else body in case there are no braces
+
+; try/catch
+(try_statement) @scope ; covers try+catch, individual try and catch are covered by (block)
```

Comment:
Why is the whole `try_statement` necessary if individual blocks are covered ?

Reply:
to grab the whole exception handling logic - [this](https://imgur.com/z6pjf8Y) would be nice to have